doc/optimizers.md

```diff
@@ -1394,6 +1394,114 @@ double Optimize(arma::mat& X);
  * [Semidefinite programming on Wikipedia](https://en.wikipedia.org/wiki/Semidefinite_programming)
  * [Semidefinite programs](#semidefinite-programs) (includes example usage of `PrimalDualSolver`)
 
+## Quasi-Hyperbolic Momentum Update
+
+ *An optimizer for [differentiable separable functions](#differentiable-separable-functions).*
+
+ Quasi Hyperbolic Momentum Update is an update policy for SGD where the Quasi Hyperbolic terms are added to the
+ parametrisation. Simply put QHM’s update rule is a weighted average of momentum’s and plain SGD’s
+ update rule.
+
+#### Constructors
+
+  * `QHSGD()`
+  * `QHSGD(`_`stepSize, batchSize`_`)`
+  * `QHSGD(`_`stepSize, batchSize, maxIterations, tolerance, shuffle`_`)`
+
+ Note that `QHSGD` is based on the templated type
+ `SGD<`_`UpdatePolicyType, DecayPolicyType`_`>` with _`UpdatePolicyType`_` =
+ QHUpdate` and _`DecayPolicyType`_` = NoDecay`.
+
+#### Attributes
+
+ | **type** | **name** | **description** | **default** |
+ |----------|----------|-----------------|-------------|
+ | `double` | **`stepSize`** | Step size for each iteration. | `0.01` |
+ | `size_t` | **`batchSize`** | Batch size to use for each step. | `32` |
+ | `size_t` | **`maxIterations`** | Maximum number of iterations allowed (0 means no limit). | `100000` |
+ | `double` | **`tolerance`** | Maximum absolute tolerance to terminate algorithm. | `1e-5` |
+ | `bool` | **`shuffle`** | If true, the function order is shuffled; otherwise, each function is visited in linear order. | `true` |
+
+ Attributes of the optimizer may also be modified via the member methods
+ `StepSize()`, `BatchSize()`, `MaxIterations()`, `Tolerance()`, and `Shuffle()`.
+
+ Note that the `QHUpdate` class has the constructor  `QHUpdate(`_`v, momentum`_`)` with a
+ default value of `0.7` for the quasi hyperbolic term and `0.999` for the momentum term.
+
+#### Examples
+
+ ```c++
+ RosenbrockFunction f;
+ arma::mat coordinates = f.GetInitialPoint();
+
+ QHSGD optimizer(0.01, 32, 100000, 1e-5, true);
+ optimizer.Optimize(f, coordinates);
+ ```
+
+#### See also:
+
+  * [Quasi-Hyperbolic Momentom and Adam For Deep Learning](https://arxiv.org/pdf/1810.06801.pdf)
+  * [Momentum SGD](#momentum-sgd)
+  * [Nesterov Momentum SGD](#nesterov-momentum-sgd)
+  * [SGD in Wikipedia](https://en.wikipedia.org/wiki/Stochastic_gradient_descent)
+  * [Differentiable separable functions](#differentiable-separable-functions)
+
+## QHAdam
+
+ *An optimizer for [differentiable separable functions](#differentiable-separable-functions).*
+
+ QHAdam is an optimizer which implements the QHAdam Adam algorithm
+ which uses Quasi-Hyperbolic Descent with the Adam Optimizer. This Method is the Adam Variant of the Quasi -
```

Comment:
Should be "This method is the Adam variant of the quasi-hyperbolic update for Adam".